scipy/optimize/tests/test_linprog.py

```diff
@@ -2212,12 +2212,12 @@ def test_mip1(self):
         # solve non-relaxed magic square problem (finally!)
         # also check that values are all integers - they don't always
         # come out of HiGHS that way
-        n = 4
+        n = 3
```

Comment:
I don't remember this being outrageously slow. How long does it take without these changes? 
This is all I am qualified to review in this PR.

Reply:
This was causing the CI to timeout.  I remember this happening before on other PRs, but I can't remember which ones off the top of my head.  Is the timeout on CI an actual bug or is it fine to reduce the complexity of the problem?

Reply:
I think this is the PR I was remembering where we saw the >=60 seconds timeouts: gh-16814

Reply:
I'll have to check, but it might be a regression. I don't remember the test as being super slow.

Reply:
I checked out this branch, merged main, updated the git submodule, and built. I ran the offending test before (with main) and after (with this PR) both with `n=4`. 
Before: 4.23s
After: 115.33s
Can you check the same problem with HiGHS directly and, if that confirms the regression, report it?

Reply:
> I think this is the PR I was remembering where we saw the >=60 seconds timeouts: https://github.com/scipy/scipy/pull/16814

I think it was the test in the PR that was timing out.

Reply:
@mdhaber This regression appears to be resolved by pulling in the latest upstream

Reply:
OK. In that case, as long as tests pass, I don't think there's anything for me to review here.